src/reader.rs

```diff
@@ -606,3 +602,37 @@ pub fn item_leaf<'a, D: Distance>(
         None => Ok(None),
     }
 }
+
+pub fn median_based_top_k<T>(v: Vec<T>, k: usize, mut threshold: T) -> Vec<T>
+where
+    T: Ord + Copy,
+{
+    let mut buffer = Vec::with_capacity(2 * k.max(1));
```

Comment:
```suggestion
pub fn median_based_top_k<T>(v: Vec<T>, k: usize) -> Vec<T>
where
    T: Ord + Copy,
{
    let mut threshold = u32::MAX;
    let mut buffer = Vec::with_capacity(2 * k.max(1));
```

Could we just move the thresholds in there?
I don't see any use cases in arroy where we would want to specify something else than MAX 🤔 

Also, for the same reason, is it really useful to be generic over `T`? The code is already hard to understand and I would like to simplify it as much as possible 😅

Reply:
I think I initially made it generic cause I was trying out different alternatives for `OrderedFloat` in another PR (spoiler: I think we can get rid of the `ordered_float` dependency and speed up some binary heap ops by 2% 👀) and just needed a top k function that worked on any type that was `Ord` as I was testing stuff.

For this PR though I agree with you, I think its better just to keep it simple & readable so I'll revert that :-)